[lilyium] iP

lilyium opened the pull request:
# Baron
> "The lowest rank of nobility"

Baron frees your mind of having to remember things you do. It's
- text-based
- easy to learn
- ~~NOT ROBUST~~

All you need to do is,
1. Download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. Double-click it.
3. Add your tasks.
4. Let it manage your tasks for you :sunglasses:

And it is **FREE**!

Features:
- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Baron("./data/baron.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

lilyium committed:
Complete Level 0

lilyium committed:
Complete Level 1

lilyium committed:
Complete Level 2

lilyium committed:
Complete Level 3

lilyium committed:
Complete Level 4

lilyium committed:
Complete Automated Text UI Testing

lilyium committed:
Complete Level 5

lilyium committed:
Complete Level 6

lilyium committed:
Complete Level 7

lilyium committed:
Merge branch 'branch-Level-7'

lilyium committed:
Complete Level 8

lilyium committed:
Merge branch 'branch-Level-8'

lilyium committed:
Complete A-MoreOOP

lilyium committed:
Complete A-Packages

lilyium committed:
Merge branch 'A-Packages'

lilyium committed:
Merge branch 'add-gradle-support'

lilyium committed:
Complete A-Gradle

lilyium committed:
Merge branch 'A-Gradle'

lilyium committed:
Improve handling of invalid inputs

lilyium committed:
Merge branch 'better-exception-handling'

wowwwp commented on src/main/java/baron/Ui.java:
```diff
@@ -0,0 +1,95 @@
+
+public class Ui {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private static final String WELCOME_MSG = "Hello! I'm Baron.\nWhat can I do for you?";
```
I like how you named your variables here

wowwwp commented on src/main/java/baron/Parser.java:
```diff
@@ -0,0 +1,195 @@
+import baron.task.Task;
+import baron.task.ToDoTask;
+
+import java.time.LocalDateTime;
```
I believe the java imports should come before the baron imports

wowwwp commented on src/main/java/baron/Storage.java:
```diff
@@ -0,0 +1,57 @@
+        return taskList;
+    }
+
+    public void saveTasks(ArrayList<Task> taskList) {
```
The method names are very clear and descriptive

wowwwp commented on src/main/java/baron/task/Task.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @Override
+    public String toString() {
```
Any reason why you decided to use an if-else block instead of handling the logic in another function?

wowwwp reviewed (commented):
Looks good!

lilyium committed:
Complete A-Junit and A-Jar

lilyium committed:
Merge branch 'A-Junit'

lilyium committed:
Complete A-JavaDoc

lilyium committed:
Complete A-CodingStandard

lilyium committed:
Merge branch 'branch-A-JavaDoc'

lilyium committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/baron/Parser.java

lilyium committed:
Complete Level 9

lilyium committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/baron/Ui.java
#	src/main/java/baron/task/Task.java

chenxy12345 commented on src/main/java/baron/Parser.java:
```diff
@@ -0,0 +1,195 @@
+    }
+
+    public static LocalDateTime parseDateTime(String dateTimeString) throws InvalidDateTimeException {
+        DateTimeFormatter dateTimeFormatter = new DateTimeFormatterBuilder()
```
Good formatting of the long statement

chenxy12345 commented on src/test/java/baron/UiTest.java:
```diff
@@ -0,0 +1,74 @@
+package baron;
```
I like that you are consistent in your ordering of import statements of the different packages across all classes

chenxy12345 commented on src/main/java/baron/Ui.java:
```diff
@@ -0,0 +1,95 @@
+public class Ui {
+    private static final String HORIZONTAL_LINE = "____________________________________________________________";
+    private static final String WELCOME_MSG = "Hello! I'm Baron.\nWhat can I do for you?";
+    private static final String GOODBYE_MSG = "Bye. Hope to see you again soon!";
```
Real. I like how your variable names are very clear

chenxy12345 commented on src/main/java/baron/exception/WrongUsageException.java:
```diff
@@ -0,0 +1,28 @@
+package baron.exception;
+
+import baron.command.Command.CommandType;
+
```
I like the naming of your exceptions. It is clear and conveys its purpose effectively.

lilyium committed:
Complete A-CheckStyle

Fix style violations detected by Checkstyle

lilyium committed:
Merge branch 'branch-A-CheckStyle'

lilyium committed:
Add JavaFX dependencies

lilyium committed:
Merge branch 'branch-Level-10'

lilyium committed:
Add GUI template

Chatbot has a text UI

A proper GUI is more user friendly and visually appealing

Let's implement a GUI using JavaFX

Using JavaFX is preferable as it allows for a platform independent
chatbot

lilyium committed:
Fix variable naming and Javadoc comments

lilyium committed:
Change methods that generate messages to return a String instead of
printing to console

Methods that generate messages print them to console and return void

Implementing a GUI requires messages to be passed along as strings to be
displayed in the interface

Let's change the method signatures to return a String of the message

lilyium committed:
Delete methods and files made unnecessary by the implementation of a GUI

lilyium committed:
Implement feature to close chatbot when user says "bye"

lilyium committed:
Merge branch 'branch-Level-10'

lilyium committed:
Complete A-Assertions

lilyium committed:
Complete A-CodeQuality

lilyium committed:
Merge pull request #2 from lilyium/branch-A-Assertions

Complete A-Assertions

lilyium committed:
Merge branch 'master' into branch-A-CodeQuality

lilyium committed:
Merge pull request #3 from lilyium/branch-A-CodeQuality

Complete A-CodeQuality

lilyium committed:
Complete C-FriendlierSyntax

lilyium committed:
Merge branch 'C-FriendlierSyntax'

lilyium committed:
Complete A-UserGuide

lilyium committed:
Merge branch 'A-UserGuide'

lilyium committed:
Place README in /docs folder

lilyium committed:
Improve UI

Let's implement level 5 of the JavaFX tutorial.

lilyium committed:
Merge branch 'better-UI'

lilyium committed:
Merge branch 'master' into branch-A-JavaDoc

lilyium committed:
Add Javadoc for most public classes and methods

lilyium committed:
Merge branch 'branch-A-JavaDoc'

lilyium committed:
Merge branch 'master' into branch-A-Junit

lilyium committed:
Fix Junit tests

Some Junit tests are broken due to changing the return type of methods
in the Ui class when completing level 10.

lilyium committed:
Merge branch 'branch-A-Junit'

lilyium committed:
Add product screenshot to README
